utilities/claim/claim.js

```diff
@@ -160,49 +168,64 @@ async function initiateRound(privateKey, { ethRegistryAddress, ethTokenAddress,
       gas = await ethRewardsManagerContract.methods.transferToSolana(arbiterFee, nonce).estimateGas()
       const tx = await ethRewardsManagerContract.methods.transferToSolana(arbiterFee, nonce).send({
         from: accountAddress,
-        gas
+        gas,
+        gasPrice
       })
 
       const txHash = tx.transactionHash
       console.log('Successfully transferred to wormhole, transaction:', txHash)
+      // Append transaction hash to file
+      fs.appendFileSync(wormholeFile, txHash + '\n')
+    }
 
-      console.log('Redeeming wormhole funds on Solana')
-      const wh = await wormhole("Mainnet", [evm, solana])
-      const sendChain = wh.getChain("Ethereum")
-      const receiveChain = wh.getChain("Solana")
-      const tokenBridge = await receiveChain.getTokenBridge()
-      const signer = await getWormholeSolanaSigner(receiveChain, feePayerSecretKey, solanaRpcEndpoint)
-      const [whm] = await sendChain.parseTransaction(txHash)
-
-      const oneHourInMs = 3600000
-
-      let vaa = await wh.getVaa(whm, "TokenBridge:Transfer", oneHourInMs)
-      let retryCount = 0
-      while (!vaa) {
-        logger.warn(
-          { tx: txHash, retryCount: retryCount++ },
-          "No TokenBridgeTransfer VAA found, waiting and trying again..."
-        );
-        await new Promise((resolve) => setTimeout(resolve, 1000))
-        vaa = await wh.getVaa(whm, "TokenBridge:Transfer", oneHourInMs)
-      }
-      const isComplete = await tokenBridge.isTransferCompleted(vaa)
-      console.log(`Wormhole redemption completion status: ${isComplete}`)
-      // Wait for 30 seconds to ensure the VAA is processed
-      await new Promise((resolve) => setTimeout(resolve, 30_000))
-      if (!isComplete) {
-        const redeemTxs = tokenBridge.redeem(signer.address.address, vaa)
-        const resRedeem = await signSendWait(
-          receiveChain,
-          redeemTxs,
-          signer.signer
-        )
-      }
-      console.log('Wormhole redemption completed')
-
+    // Read and process all pending transactions
+    const transactions = fs.readFileSync(wormholeFile, 'utf8').split('\n').filter(Boolean)
```

Comment:
Related to above, this might throw if the file doesn't exist?

Reply:
nice